opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -371,6 +372,17 @@ def detect(self) -> "Resource":
         return Resource(resource_info)
 
 
+class ServiceInstanceIdResourceDetector(ResourceDetector):
+    # pylint: disable=no-self-use
+    _instance_id_cache = {}
+
+    def detect(self) -> "Resource":
```

Comment:
Any reason not to simply wait until these get added as `stable` and include them in `OtelResourceDetector`?

Reply:
> Are there plans to include them as part of "Semantic Attributes with SDK-provided Default Value"

Yes, see https://github.com/open-telemetry/semantic-conventions/issues/311.

> Any reason not to simply wait until `service.instance.id` get added as stable and include them in `OtelResourceDetector`?

One prerequisite for adding `service.instance.id` as stable are working prototypes and implementation in several languages. Having such an implementation (or at least prototype) in Python would be beneficial for those stabilization efforts, especially as the description of `service.instance.id` explicitly mentions application servers like gunicorn.

Showing that `service.instance.id` can be populated according to the conventions for all major usage scenarios in Python would be a prerequisite for declaring it stable and making it a default attribute.

Reply:
> Yes, see https://github.com/open-telemetry/semantic-conventions/issues/311.

> Having such an implementation (or at least prototype) in Python would be beneficial for those stabilization efforts

If there are plans to make them default doesn't that mean they would move into `OtelResourceDetector` eventually? If we create a new resource detector (even though it might remain experimental) it sounds like we would have to support this forever. Maybe I might be misunderstanding what "prototype" is, will this just be temporary to get the specs to move along?